doc/api/assert.md

```diff
@@ -137,13 +137,11 @@ const assert = require('assert');
 ```
 
 Whenever possible, use the [`strict` mode][] instead. Otherwise, the
-[Abstract Equality Comparison][] may cause surprising results. This is
-especially true for [`assert.deepEqual()`][], where the comparison rules are
-lax:
+[Abstract Equality Comparison][] may cause surprising results.
 
 ```js
 // WARNING: This does not throw an AssertionError!
-assert.deepEqual(/a/gi, new Date());
```

Comment:
i think it would be very surprising to change this. One of the whole points of this method is to compare distinct objects based on their shape/contents and not based on their identity.

Reply:
That was just an somewhat unrelated update to the docs. It does throw an error by now and the example is outdated.

Reply:
ok, fair on this line - but my comment still applies to “reference equality in objects”. It’s trivial to do that already with ===; what’s hard is loose equality, which is what the non-strict assert methods do.

Reply:
The deep equal function is not changed at all and it still applies the loose equality check while comparing properties. This mainly aligns the loose equal check with the deep equal one by special handling objects. Comparing objects using the abstract equality calls `toString` and that definitely causes surprising behavior. This is just a step to limit the surprising parts a tiny bit. It's definitely not ideal but it might be some kind of compromise for the loose equality check.

Reply:
That's not surprising, that's the purpose of loose equality checks when comparing an object and a non-object. It's a desired feature of deepEqual.